[13.0] Add edi_gs1 + edi_gs1_stock

acsonefho opened the pull request.

simahawk committed:
Add edi_gs1

simahawk committed:
Add edi_gs1_stock

simahawk committed:
edi_gs1: fix  output tests

simahawk committed:
edi_gs1_stock: adapt to edi core changes

simahawk committed:
edi_gs1_stock: fix inbound_instruction

simahawk committed:
edi_gs1: BH instance identifier = record identifier

simahawk committed:
edi_gs1: add basic types for ack

simahawk committed:
edi_gs1: add basic component for output

simahawk committed:
edi_gs1: add schemas for warehouse exchanges

simahawk committed:
edi_gs1_stock: inbound info based on output mixin

simahawk committed:
edi_gs1_stock: inbound instruction fix identification

simahawk committed:
edi_gs1_stock: inbound instruction use ack type

simahawk committed:
edi_gs1: draft outbound instruction

simahawk committed:
edi_gs1: clean dead code

simahawk committed:
edi_gs1: depend on edi_xml

simahawk committed:
edi_gs1_stock: add xml validation config

simahawk committed:
edi_exchange_template: use type conf if available

simahawk committed:
edi_gs1: component has only exc record in ctx

simahawk committed:
edi_gs1_stock: adapt to edi api changes

simahawk committed:
edi_gs1: SBH tests validate xml

simahawk committed:
edi_gs1_stock: tests validate xml

simahawk committed:
edi: draft gs1 input mixin

simahawk committed:
edi_exchange_template: draft input template

simahawk committed:
edi_exchange_template: include template in render ctx

simahawk committed:
edi_exchange_template: add TODO

simahawk committed:
edi_gs1_stock: change inbound instruction component name

simahawk committed:
edi_gs1_stock: fix inbound notif test class name

simahawk committed:
edi_gs1_stock: draft notification components

simahawk committed:
DROPME: add TODOs

simahawk committed:
edi_gs1: adapt tests to core changes

simahawk committed:
edi_exchange_template: fix info provider lookup

simahawk committed:
edi_exchange_template: rework info provider mixins

simahawk committed:
edi_gs1_stock: fix inbound/outbound instruction tests

simahawk committed:
edi_gs1_stock: TEMP disable tests for inbound notification

acsonefho committed:
[ADD] edi_exchange_template: make the env available into context

acsonefho committed:
[ADD] edi_gs1: add gs1_code on UoM (+ auto init based on default values) + add a contact details qweb template

acsonefho committed:
[UPD] edi_gs1_stock: update outbound instruction qweb to add contact details + simulate a fake res.partner record into template

simahawk committed:
edi_gs1: add gs1_code to UoM

GS1 standard requires UoM specific mapping.
Here we add a specific field for it
and provide mapping of values
as well as auto compute for existing UoM.

simahawk committed:
edi_gs1_stock: improve outbound instruction

* Add qweb tmpl for contact details
* Fix fake res.partner record for shipper

simahawk committed:
edi_storage: draft receive component

simahawk committed:
edi_xml: raise EDIValidationError

simahawk committed:
edi_gs1: split ack data type + imp conf

simahawk committed:
edi_gs1: draft ack input process component

acsonefho committed:
[IMP] edi: file name builder

acsonefho committed:
[ADD] edi_gs1: lsp and lsc partners on backend

acsonefho committed:
[FIX] edi_gs1: empty doc_type invalidate the document generation

acsonefho committed:
[UPD] edi_gs1: update group on security

acsonefho committed:
[IMP] edi_gs1_stock: pre-commit

acsonefho committed:
[UPD] edi_gs1_stock: Outbound instruction

Update outbound instruction + related components + adapations also in Inbound instruction + adapt tests

acsonefho committed:
Pre-commit

acsonefho committed:
[FIX] Fix unit test about output generation + missing security rules

acsonefho committed:
[FIX] edi: fix minor issues.

acsonefho committed:
[UPD] GS1 .xsd files update

acsonefho committed:
[FIX] edi_gs1: fix typo and find_component.

acsonefho committed:
[UPD] edi_storage: filename_pattern builder update

acsonefho committed:
[FIX] edi_storage: fix typo

acsonefho committed:
[ADD] edi_gs1_stock: enable Warehouse Outbound Notification

simahawk committed:
edi_gs1_stock: DRAFT input

acsonefho committed:
[FIX] edi: fix due to refactor

agent-z28 commented:
NICE

agent-z28 commented:
@acsonefho i think the edi_gs1 module should only extend the parner models etc, and be morge generic. 
If we only want edi_gs1_edifact integration all the XML stuff will be unwanted. but the gln is needed, and so on

simahawk commented:
I think we should merge it as it is and refactor things out later if needed.

simahawk commented:
of course, we need reviews to merge it :wink:

agent-z28 commented:
@simahawk it's part of my review, that i implement another edi_gs1 standard, and the edi_gs1 here is not compatible. so i cannot use it like this. the module should only contain edi_gs1 related things. the xml implementatin should be edi_gs1_xmlstandardname while i do edi_gs1_edifact_*.

simahawk commented:
what I'm saying is that if what we have here works it would be nice to merge it and then you can propose any change you think it would make it better. 
So, could you please drop a review for what it is today and just ask to add a line on the roadmap to split out the xml part from edi_gs1? It will be easier for everybody and for us especially to review the changes.
Could it work for you?

agent-z28 commented:
@simahawk i understand that it would be easier, but afterwards, you need migration scripts and it'll create much more trouble as splitting it directly. i can add pull request with edi_gs1 modul on its own?

simahawk commented:
> @simahawk i understand that it would be easier, but afterwards, you need migration scripts and it'll create much more trouble as splitting it directly. i can add pull request with edi_gs1 modul on its own?

sure, feel free :)

etobella commented:
@agenterp Any news on your proposal? If it is not possible to have the new changes, we can start with this version and then propose the improvements.

agent-z28 commented:
@etobella i'm still working on this issue internally. will be done in the middle of next week. But starting with a module name, where we later need migrations, and we already know it, seems not be a good approach. another general question, why has the edi module be renamed to edi_oca? i thought company or organisation names doesn't belong to module names.

etobella commented:
`edi` was taken by odoo and was not published on odoo apps (but there was no module...). At the end, we decided to change the name in order to erase problems. (renaming is done by OpenUpgrade, so not a problem on migration)

About migrations. it is not really a problem for me, at the end, a lot of migrations are made on version change every year or even in the same version, so I don't see the issue there

agent-z28 commented:
@etobella @simahawk  okay if you dont see a problem in migration later, i will not block it, i just wanted to mention, that i already have the problem. but if it's easy to fix it later lets go

bealdav commented:
Hi sorry for my noob question but I would like to know if you plan or need a future plan to interact with these features
https://github.com/OCA/stock-logistics-barcode/pull/387/files 
or maybe be not because I'm really noob on this matter :-)
Maybe at least base_gs1_barcode ?
Sorry for my confuse question

simahawk commented:
@bealdav on my side no plan but I'm not working on this subject since a while. Would be nice to do it tho. Could you create an issue for this?
FTR regarding this PR, I still have to get back to it and clean it up.
I started it here https://github.com/OCA/edi/pull/449 but got no time to move it fwd.

acsonefho committed:
[ADD] Add configuration

acsonefho committed:
[IMP] Improvement to create the ACK

acsonefho committed:
[INT] pre-commit

acsonefho committed:
[GS1] Generate the Warehouse Outbound Notification ACK

acsonefho committed:
[GS1] Add shipper on WOI + add inherit point on config view + fix template

acsonefho committed:
[EDI] Force re-generate filename during generation of EDI content

acsonefho committed:
[FIX] edi_gs1: remove useless _xsd_schema_path

acsonefho committed:
[UPD] edi: automatically retry when file not on storage (input)

acsonefho committed:
[UPD] Add possibility to stop the GS1 Workflow on the stock.picking

acsonefho committed:
[EDI][GS1] Avoid post message with same content.
That could happens when a job is checking if a file must be processed/read/... And there is a lot of mail.message generated with the same information

acsonefho committed:
[EDI] Avoid create multiple job for same exchange record with same state

bosd commented:
Has  this been  included in a different pr?

acsonefho commented:
> Has this been included in a different pr?

Unfortunately not :/

simahawk commented:
I'll try to have a look at this by the end of the year, even if is a bit hard because I'm not working on v13 since a while.
Plus, there are a lot of changes that should be backported from 14 that might be tricky to integrate.

bosd commented:
Any update on this :shy:?

acsonefho commented:
> Any update on this :shy:?

Probably (and certainly) some improvements must be done.
Already in production for almost 2 years and working perfectly.
So no update because "it's working" :muscle:

bosd commented:
You're still running some install on V13?
I have some v14 in production and feel like it's time to migrate..
